cibuildwheel/windows.py

```diff
@@ -230,6 +255,43 @@ def setup_python(
     return env
 
 
+def fix_cross_compiled_wheel(target_arch: str, wheel: Path) -> None:
+    # replace occurrences to win_amd64 by win_{target_arch} in wheel
+    #
+    # in theory, set SETUPTOOLS_EXT_SUFFIX=.cp310-win_arm64.pyd should be able
+    # to change name of .pyd files when building wheel.
+    # Alas, it requires using distutils from setuptools
+    # (SETUPTOOLS_USE_DISTUTILS=local), which is not supported on all projects.
+    # Thus, we change name of pyd files here.
```

Comment:
The correct fix is to change `sysconfig.get_config_var('EXT_SUFFIX')` / `distutils.sysconfig.get_config_var('EXT_SUFFIX')`, I believe - this should work for everyone.

Reply:
Changing sysconfig stuff implies to change project targeted. I'm not sure it's feasible for numpy since they override those things. In more, only "future" versions would be compatible. Patching the wheel is dirty, but it provides an applicable solution. As said in comment, there is already a builtin env var supporting this in setuptools, so we can drop wheel patching once concerned projects are all "compatible". Do you have another idea for a cleaner and applicable alternative?

For crossenv, it's only supporting Linux for now (it seems there was [interest](https://discuss.python.org/t/towards-standardizing-cross-compiling/10357/3) for porting to WIndows, but it was not made).